service/topology_coordinator.cc

```diff
@@ -681,6 +687,51 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         case global_topology_request::cleanup:
             co_await start_cleanup_on_dirty_nodes(std::move(guard), true);
             break;
+        case global_topology_request::keyspace_rf_change: {
+            while (true) {
+                rtlogger.info("raft topology: new keyspace RF change requested");
+                auto tmptr = get_token_metadata_ptr();
+                sstring ks_name = *_topo_sm._topology.new_keyspace_rf_change_ks_name;
+                std::map<sstring, sstring> new_rf_per_dc = *_topo_sm._topology.new_keyspace_rf_change_rf_per_dc;
+                std::vector<canonical_mutation> updates;
+                for (const auto& table : _db.find_keyspace(ks_name).metadata()->tables()) {
+                    auto new_tablet_map = co_await reallocate_tablets_for_new_rf(table, tmptr, new_rf_per_dc);
+                    rtlogger.debug("Updating tablet map for {}.{}", ks_name, table->cf_name());
+                    auto tablet_id = new_tablet_map.first_tablet();
+                    while (true) {
+                        auto& tablet_info = new_tablet_map.get_tablet_info(tablet_id);
+                        auto last_token = new_tablet_map.get_last_token(tablet_id);
+
+                        updates.emplace_back(replica::tablet_mutation_builder(guard.write_timestamp(), table->id())
+                                                                     .set_replicas(last_token, tablet_info.replicas)
+                                                                     .set_stage(last_token, locator::tablet_transition_stage::allow_write_both_read_old)
+                                                                     .set_transition(last_token, locator::tablet_transition_kind::rf_change)
+                                                                     .build());
+                        updates.push_back(canonical_mutation(topology_mutation_builder(guard.write_timestamp())
```

Comment:
This can be added once for all tablets since it changes global state, not per-tablet state.

Reply:
OK, moved.